src/java_bytecode/java_types.h

```diff
@@ -213,6 +213,20 @@ class java_generic_typet:public reference_typet
     return (generic_type_argumentst &)(
       add(ID_type_variables).get_sub());
   }
+
+  /// Replace every occurrence of an argument with a given type.
+  /// \param argument generic type argument that should be replaced
+  /// \param type new type that the argument should be replaced with
+  void replace(const typet &argument, const typet &type)
+  {
+    PRECONDITION(is_reference(type));
+    auto isArgument = [&](reference_typet t) { return (t == argument); };
+    std::replace_if(
+      generic_type_arguments().begin(),
```

Comment:
Where has this come from? `generic_type_arguments` is not an input to this method, but this method doesn't belong to  a class?

Reply:
This whole function `replace` is in class `java_generic_typet`. The `generic_type_arguments()` function is in the same class.

Reply:
Ah I know what confused me - method body inside the class definition, would prefer since not completely trivial to move into the source file.